### PR TITLE
Makefile: Replace vbatts/pandoc with a PANDOC variable

### PR DESCRIPTION
**Pull request by wking**
And use `command -v` instead of `which`.  Details in the commit messages.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 
-DOCKER ?= $(shell which docker)
+DOCKER ?= $(shell command -v docker)
```
Comment (vbatts) [inline Makefile]: I suppose this is fine.

**Comment by vbatts** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (hqhq) [inline Makefile]: @wking It looks like it should work fine, but I got error `make: command: Command not found`, I have to change it back to `which`, any insight on this? Maybe related to bash version?

File: Makefile [hunk 2]
```diff
@@ -32,7 +33,7 @@ output/docs.pdf: $(DOC_FILES)
 	-v $(shell pwd)/:/input/:ro \
 	-v $(shell pwd)/output/:/output/ \
 	-u $(shell id -u) \
-	vbatts/pandoc -f markdown_github -t latex -o /$@ $(patsubst %,/input/%,$(DOC_FILES))
+	$(PANDOC) -f markdown_github -t latex -o /$@ $(patsubst %,/input/%,$(DOC_FILES))
```
Comment (laijs) [inline Makefile]: I'm just curious before any test or investigation, $(PANDOC) is a **path to the pandoc executive** when default, but we expect **pandoc image** here, how this replacement works?

Comment (wking) [inline Makefile]: On Thu, May 19, 2016 at 08:40:21PM -0700, Lai Jiangshan wrote:
> > -	vbatts/pandoc -f markdown_github -t latex -o /$@ $(patsubst %,/input/%,$(DOC_FILES))
> > +	$(PANDOC) -f markdown_github -t latex -o /$@ $(patsubst %,/input/%,$(DOC_FILES))
> 
> I'm just curious before any test or investigation, $(PANDOC) is a
> **path to the pandoc executive** when default, but we expect
> **pandoc image** here, how this replacement works?

Oops, I had missed this.  See #440 for some paths forward.

**Pull request merged**